index.html

```diff
@@ -538,20 +616,32 @@
         showSelectPlayerModal(true);
       }
       else {
-        recordMove();
+        // Tree will be harvested if it's number 4
+        var scoringToken = currCell.tree == 4 ? getScoringToken(calculateRingNumber()) : null;
+        recordMove(scoringToken);
+        var cellPlayer = currCell.player;
         currCell.growTree();
         updateTick();
       }
     }
 
-    function recordMove() {
+    // Takes a scoring token when the move will harvest a tree so the token can be replaced
+    // if the move is undone.
+    function recordMove(scoringToken) {
       var cell = currCell;
       var oldPlayer = cell.player;
       var oldTree = cell.tree;
+      if (scoringToken !== null) {
+        playerScores[oldPlayer] += scoringToken.score;
+      }
 
       moves.push(() => {
         cell.player = oldPlayer;
         cell.tree = oldTree;
+        if (scoringToken !== null) {
+          playerScores[oldPlayer] -= scoringToken.score;
```

Comment:
i think to be consistent we can have a `oldScore` variable

Reply:
Fixed. I also moved the score addition outside of `recordMove`. Think it's neater to have the score addition be done in `toggleCell` instead.